### PR TITLE
eslint config update [ skip builds ]

### PR DESCRIPTION
**Pull request by thoys**
- Sorted globals / rules
- Added missing globals
   - Agent (assignment-client scripts)
   - AudioScope
   - AudioStats
   - Avatar (assignment-client scripts)
   - AvatarBookmarks
   - AvatarInputs
   - Desktop
   - DesktopPreviewProvider
   - document
   - EntityViewer (assignment-client scripts)
   - EventBridge
   - GooglePoly
   - Graphics
   - LocationBookmarks
   - Midi
   - OffscreenFlags
   - OverlayWindow
   - Picks
   - PickType
   - PointerEvent
   - Pointers
   - QmlFragment
   - Render (interface)
   - Selection
   - Snapshot
   - SpeechRecognizer
   - Steam
   - Users (all contexts' except ESS)
   - Wallet
- Made location global write-able since you could teleport by assigning another hifi:// url to that global e.g. `location = hifi://hq`
- Added rules:
   - `key-spacing`, to prevent inconsistencies when describing objects.  This rule allows `{type: "Cube"}` but prevents `{type : "Cube"}` and `{type:"Cube"}`
   - `space-before-blocks`  is consistent with the examples of the styleguide e.g. [4.1.5](https://docs.highfidelity.com/build-guide/coding-standards#4-1-5-the-if-else-class-of-statements-should-have-the-following)
  - `no-magic-numbers` with the exceptions -1, 0, 0.5, 1 and 2

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -45,40 +56,36 @@ module.exports = {
...
+        "indent": ["error", 4, {"SwitchCase": 1}],
+        "key-spacing": ["error", {"beforeColon": false, "afterColon": true, "mode": "strict"}],
+        "keyword-spacing": ["error", {"before": true, "after": true}],
         "max-len": ["error", 128, 4],
         "new-cap": ["error"],
+        "no-console": ["off"],
         "no-floating-decimal": ["error"],
-        //"no-magic-numbers": ["error", { "ignore": [0, 1], "ignoreArrayIndexes": true }],
-        "no-multiple-empty-lines": ["error"],
+        // "no-magic-numbers": ["error", {"ignore": [0, 1], "ignoreArrayIndexes": true}],
```
Comment (elisa-lj11) [inline .eslintrc.js]: delete comment unless you think we may use this later

File: .eslintrc.js [hunk 2]
```diff
@@ -6,34 +6,45 @@ module.exports = {
...
+        "console": false,
+        "ContextOverlay": false,
         "Controller": false,
-        "DialogsManager": false,
         "DebugDraw": false,
+        "DialogsManager": false,
         "Entities": false,
+        "EntityViewer": false,
         "FaceTracker": false,
         "GlobalServices": false,
```
Comment (elisa-lj11) [inline .eslintrc.js]: Where's "GooglePoly"?

File: .eslintrc.js (hunk 1 above)
Comment (thoys) [reply .eslintrc.js]: it was already in the original file, I'm curious why it was commented out too..

File: .eslintrc.js [hunk 3]
```diff
@@ -6,34 +6,45 @@ module.exports = {
     },
     "globals": {
         "Account": false,
+        "Agent": false,
         "AnimationCache": false,
         "Assets": false,
         "Audio": false,
         "AudioDevice": false,
         "AudioEffectOptions": false,
```
Comment (MiladNazeri) [inline .eslintrc.js]: AudioScope, AudioStats

File: .eslintrc.js [hunk 4]
```diff
@@ -6,34 +6,45 @@ module.exports = {
...
+        "Avatar": false,
         "AvatarList": false,
```
Comment (MiladNazeri) [inline .eslintrc.js]: AvatarBookmarks

Comment (MiladNazeri) [reply .eslintrc.js]: AvatarInputs

File: .eslintrc.js [hunk 5]
```diff
@@ -6,34 +6,45 @@ module.exports = {
...
         "AvatarManager": false,
         "Camera": false,
         "Clipboard": false,
+        "console": false,
```
Comment (MiladNazeri) [inline .eslintrc.js]: Is this Console or console and does it matter?

File: .eslintrc.js [hunk 6]
```diff
@@ -6,34 +6,45 @@ module.exports = {
...
+        "ContextOverlay": false,
         "Controller": false,
```
Comment (MiladNazeri) [inline .eslintrc.js]: Desktop

Comment (MiladNazeri) [reply .eslintrc.js]: DesktopPreviewProvider

File: .eslintrc.js (hunk 2 above)
Comment (MiladNazeri) [inline .eslintrc.js]: File

File: .eslintrc.js [hunk 7]
```diff
@@ -6,34 +6,45 @@ module.exports = {
...
-        "DialogsManager": false,
         "DebugDraw": false,
+        "DialogsManager": false,
         "Entities": false,
+        "EntityViewer": false,
         "FaceTracker": false,
         "GlobalServices": false,
         "HMD": false,
```
Comment (MiladNazeri) [inline .eslintrc.js]: Graphics

File: .eslintrc.js [hunk 8]
```diff
@@ -6,34 +6,45 @@ module.exports = {
...
+        "LaserPointers": false,
+        "location": true,
         "LODManager": false,
```
Comment (MiladNazeri) [inline .eslintrc.js]: LocationBookmarks

File: .eslintrc.js [hunk 9]
```diff
@@ -6,34 +6,45 @@ module.exports = {
...
         "Mat4": false,
         "Menu": false,
         "Messages": false,
         "ModelCache": false,
```
Comment (MiladNazeri) [inline .eslintrc.js]: Midi

File: .eslintrc.js [hunk 10]
```diff
@@ -6,34 +6,45 @@ module.exports = {
...
+        "module": false,
         "MyAvatar": false,
         "Overlays": false,
```
Comment (MiladNazeri) [inline .eslintrc.js]: OffscreenFlags

File: .eslintrc.js [hunk 11]
```diff
@@ -6,34 +6,45 @@ module.exports = {
...
         "OverlayWebWindow": false,
```
Comment (MiladNazeri) [inline .eslintrc.js]: OverlayWindow

File: .eslintrc.js [hunk 12]
```diff
@@ -6,34 +6,45 @@ module.exports = {
...
         "Paths": false,
+        "print": false,
```
Comment (MiladNazeri) [inline .eslintrc.js]: Picks

Comment (MiladNazeri) [reply .eslintrc.js]: PickType
PointerEvent
Pointers

Comment (MiladNazeri) [inline .eslintrc.js]: QmlFragment

File: .eslintrc.js [hunk 13]
```diff
@@ -6,34 +6,45 @@ module.exports = {
...
         "Quat": false,
         "Rates": false,
+        "RayPick": false,
         "Recording": false,
+        "Render": false,
         "Resource": false,
         "Reticle": false,
         "Scene": false,
```
Comment (MiladNazeri) [inline .eslintrc.js]: Selection

File: .eslintrc.js [hunk 14]
```diff
@@ -45,40 +56,36 @@ module.exports = {
         "Tablet": false,
```
Comment (MiladNazeri) [inline .eslintrc.js]: Snapshot

Comment (MiladNazeri) [inline .eslintrc.js]: SpeechRecognizer

Comment (MiladNazeri) [reply .eslintrc.js]: Steam

File: .eslintrc.js [hunk 15]
```diff
@@ -45,40 +56,36 @@ module.exports = {
...
         "TextureCache": false,
```
Comment (MiladNazeri) [inline .eslintrc.js]: TabletProxy
TabletButtonProxy
tabletInterface

File: .eslintrc.js [hunk 16]
```diff
@@ -45,40 +56,36 @@ module.exports = {
...
         "Toolbars": false,
-        "Uuid": false,
         "UndoStack": false,
+        "Users": false,
         "UserActivityLogger": false,
+        "Uuid": false,
         "Vec3": false,
         "WebSocket": false,
```
Comment (MiladNazeri) [inline .eslintrc.js]: Wallet

File: .eslintrc.js (hunk 5 above)
Comment (thoys) [reply .eslintrc.js]: its console, its consistent with the build in web-javascript feature Console, which is exposed as `console` , https://developer.mozilla.org/en-US/docs/Web/API/Console note that the actual object is called `Console` in their API

File: .eslintrc.js [hunk 17]
```diff
@@ -50,17 +67,22 @@ module.exports = {
...
+        "SpeechRecognizer": false,
         "Stats": false,
+        "Steam": false,
         "Tablet": false,
         "TextureCache": false,
         "Toolbars": false,
         "UndoStack": false,
-        "Users": false,
         "UserActivityLogger": false,
+        "Users": false,
```
Comment (elisa-lj11) [inline .eslintrc.js]: I like this alphabetization 😄

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9773/HighFidelity-Beta-PR13151-50f76dd4df4db78e3644d6328e19405bfcb87184.apk

File: .eslintrc.js (hunk 17 above)
Comment (thoys) [reply .eslintrc.js]: Yeah, I selected the lines in SublimeText and pressed F9 to sort them :D

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9773/HighFidelity-Beta-PR13151-50f76dd4df4db78e3644d6328e19405bfcb87184.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9773/HighFidelity-Beta-Interface-PR13151-50f76dd4df4db78e3644d6328e19405bfcb87184.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9773/HighFidelity-Beta-PR13151-50f76dd4df4db78e3644d6328e19405bfcb87184.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9773/HighFidelity-Beta-Interface-PR13151-50f76dd4df4db78e3644d6328e19405bfcb87184.exe

**Comment by thoys** [comment]
@ctrlaltdavid https://eslint.org/docs/user-guide/configuring does state `"no-console": "off",` to turn the rule off which is enabled by `"extends": "eslint:recommended",`

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9791/HighFidelity-Beta-PR13151-b20741baba03870f67cb36200f6ff5f1efadecf3.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9791/HighFidelity-Beta-PR13151-b20741baba03870f67cb36200f6ff5f1efadecf3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9791/HighFidelity-Beta-Interface-PR13151-b20741baba03870f67cb36200f6ff5f1efadecf3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9791/HighFidelity-Beta-PR13151-b20741baba03870f67cb36200f6ff5f1efadecf3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9791/HighFidelity-Beta-Interface-PR13151-b20741baba03870f67cb36200f6ff5f1efadecf3.exe

**Pull request merged**